docs/en/reference/custom-configuration.rst

```diff
@@ -45,24 +44,23 @@ Once you have your custom integration setup, you can modify it to look like the
 
     $configuration->setMetadataStorageConfiguration($storageConfiguration);
 
-    $helperSet = new HelperSet();
-    $helperSet->set(new QuestionHelper(), 'question');
-    $helperSet->set(new ConnectionHelper($connection), 'db');
-    $helperSet->set(new ConfigurationHelper($connection, $configuration));
+    $dependencyFactory = DependencyFactory::fromConnection(
+        new ExistingConfiguration($configuration),
+        new ExistingConnection($connection)
+    );
 
     $cli = new Application('Doctrine Migrations');
     $cli->setCatchExceptions(true);
-    $cli->setHelperSet($helperSet);
 
     $cli->addCommands(array(
-        new Command\DumpSchemaCommand(),
-        new Command\ExecuteCommand(),
-        new Command\GenerateCommand(),
-        new Command\LatestCommand(),
-        new Command\MigrateCommand(),
-        new Command\RollupCommand(),
-        new Command\StatusCommand(),
-        new Command\VersionCommand()
+        new Command\DumpSchemaCommand(null, $dependencyFactory),
```

Comment:
Would it make sense to change the constructor signature to either take the dependencyFactory first or omit the command name from the signature entirely?

Reply:
Flipping the arguments could make sense, but I'm not sure about it...
Removing the name can not be done, it is used bu the doctrine bundle to add the `doctrine:` prefix to the command.

Reply:
now `$dependencyFactory` is passed as first argument.

Reply:
Ah, I see. Keeping the name is fine if we need it.